test/e2e/scale.go

```diff
@@ -112,9 +112,37 @@ type ScaleSpecInput struct {
 	// It uses this machine deployment to create additional copies.
 	// Names of the MachineDeployments will be overridden to "md-1", "md-2", etc.
 	// The resulting number of worker nodes for each of the workload cluster will
-	// be MachineDeploymentCount*WorkerMachineCount (CAPI_SCALE_MACHINE_DEPLOYMENT_COUNT*CAPI_SCALE_WORKER_MACHINE_COUNT).
+	// be MachineDeploymentCount*WorkerPerMachineDeploymentCount (CAPI_SCALE_MACHINE_DEPLOYMENT_COUNT*CAPI_SCALE_WORKER_PER_MACHINE_DEPLOYMENT_COUNT).
 	MachineDeploymentCount *int64
 
+	// AdditionalClusterClassCount is the number of copies of the ClusterClasses that will be deployed.
+	// This can be used to test how Cluster API scales with a higher number of ClusterClasses.
+	// Can be overridden by variable CAPI_SCALE_ADDITIONAL_CLUSTER_CLASS_COUNT.
+	AdditionalClusterClassCount *int64
```

Comment:
```suggestion
        AdditionalClusterClassCount *int64 `env:” CAPI_SCALE_ADDITIONAL_CLUSTER_CLASS_COUNT” envDefault:”0”`
```

Not really a concern, but wanted to mention. We had to deal with an increasing number of input variables in our e2e tests, which have to be always specified or have a sensible default.

https://github.com/caarlos0/env was a pretty good solution to make an environment variable or a config variable the default value for the `SpecInput`.

With some minor changed the order of precedence can be defined as `input.AdditionalClusterClassCount -> env.CAPI_SCALE_ADDITIONAL_CLUSTER_CLASS_COUNT -> 0`, which removes the need to specify the default on the structure creation like in `test/e2e/scale_test.go`, and instead leave it to “non-standard” overrides. All is needed, to invoke `Parse` method somewhere:

```go
func Parse[T any](dst *T) error {
	LoadE2EConfig(os.Getenv("E2E_CONFIG"))

	src := new(T)
	if err := env.ParseWithOptions(src, EnvOptions); err != nil {
		return err
	}

	return mergo.Merge(dst, src)
}
```

Reply:
Thx for mentioning it! I'm not sure if we want to introduce a dependency to a privately-owned repo for that functionality